practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/HapiFhirRepository.java

```diff
@@ -33,28 +33,32 @@
      * @param practitioner the practitioner to save.
      * @return id of the saved practitioner
      */
-    public String savePractitioner(Practitioner practitioner) {
+    public String createPractitioner(final Practitioner practitioner) {
+
         // Log the request
         if (logger.isInfoEnabled()) {
             logger.info(FhirRepo.SAVE_REQUEST.message(), fhirContextWrapper.prettyPrint(practitioner));
         }
 
-        Bundle responseBundle;
+        final Bundle responseBundle;
         try {
             responseBundle = fhirContextWrapper.executeTransaction(bundle(practitioner));
         } catch (FhirServerResponseException e) {
             final String message = FhirRepo.FAILED_TO_SAVE_PRACTITIONER.message();
             throw new RestException(message, e);
         }
-        IBaseResource responseElement = extractResponseResource(responseBundle);
+        IBaseResource responseElement = fhirContextWrapper.getResourcesFrom(responseBundle, 1).get(0);
```

Comment:
Can we hide the .get(0) inside the wrapper - maybe a convenience first() type thing for single element ops?

Reply:
Done, with tests included - throughout this whole fix it felt like the end was about 10 minutes away, but the whole thing took 2 hours 15 minutes.  :(